lib/phoenix_live_view/html_algebra.ex

```diff
@@ -62,7 +62,17 @@ defmodule Phoenix.LiveView.HTMLAlgebra do
               ""
             end
 
-          {next_node, next_type, concat([prev_doc, flex_break(on_break), next_doc])}
+          # We can't use flex_break when the next_node is eex_token and it
+          # doesn't have a white space. Otherwise it would change the displayed
+          # text.
+          #
+          # text<%= @foo %>  - should not use flex_break
+          # text <%= @foo %> - should use flex_break
+          if eex_token?(next_node) and on_break == "" do
```

Comment:
@josevalim not sure if we can make `result<%= if(@row_count != 1, do: "s") %>` goes to the next line in this case because here we have two different tokens `text` and `eex` and both are two distinct algebra documents at the point.

I took the easier path and kept them at the same line in this case, but maybe there is a better workaround? 🤔

Reply:
I think this looks great actually! The only thing I would change is to keep only the call to flex_break in the conditional

Reply:
Cool. refactored as you suggested :)